mongo-utils/src/main/java/org/wso2/mongo/MongoDBDataSourceUtil.java

```diff
@@ -265,6 +277,69 @@ private static MongoClientOptions createOptions(MapValue options) {
         return builder.build();
     }
 
+    private static SSLContext initializeSSLContext(MapValue options) {
+        TrustManager[] trustManagers;
+        KeyManager[] keyManagers;
+
+        MapValue secureSocket = options.getMapValue(ConnectionParam.SECURE_SOCKET.getKey());
+
+        MapValue trustStore = secureSocket.getMapValue(ConnectionParam.TRUST_STORE.getKey());
+        String trustStoreFilePath = trustStore.getStringValue(ConnectionParam.CERTIFICATE_PATH.getKey()).getValue();
+        try (InputStream trustStream = new FileInputStream(trustStoreFilePath)) {
+            char[] trustStorePass = trustStore.getStringValue(ConnectionParam.CERTIFICATE_PASSWORD.getKey()).getValue()
+                    .toCharArray();
+            KeyStore trustStoreJKS = KeyStore.getInstance(KeyStore.getDefaultType());
+            trustStoreJKS.load(trustStream, trustStorePass);
+
+            TrustManagerFactory trustFactory =
+                    TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            trustFactory.init(trustStoreJKS);
+            trustManagers = trustFactory.getTrustManagers();
+        } catch (FileNotFoundException e) {
+            throw new MongoDBClientException("Trust store file not found for secure connections to MongoDB. " +
+                    "Trust Store file path : '" + trustStoreFilePath + "'.", e);
+        } catch (IOException e) {
+            throw new MongoDBClientException("I/O Exception in creating trust store for secure connections to " +
+                    "MongoDB. Trust Store file path : '" + trustStoreFilePath + "'.", e);
+        } catch (GeneralSecurityException e) {
+            throw new MongoDBClientException("Error in initializing certs for Trust Store : " +
+                    e.getMessage(), e.getCause());
+        }
+
+        MapValue keyStore = secureSocket.getMapValue(ConnectionParam.KEY_STORE.getKey());
+        String keyStoreFilePath = keyStore.getStringValue(ConnectionParam.CERTIFICATE_PATH.getKey()).getValue();
```

Comment:
Is it mandatory to provide a keystore?

Reply:
Yes. Since MongoDB works on mutual SSL. Both certs are mandatory